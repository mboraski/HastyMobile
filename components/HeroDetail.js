// Third Part Imports
import React from 'react';
import { Text, View, Image, StyleSheet, TouchableOpacity } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import { Icon } from 'react-native-elements';

// Relative Imports
import Color from '../constants/Color';
import Style from '../constants/Style';
import { emY } from '../utils/em';
import chatIcon from '../assets/icons/chat.png';

const IMAGE_SIZE = emY(4.25);
const CHAT_SIZE = emY(3.375);
const CHAT_IMAGE_SIZE = emY(1.3125);

const HeroDetail = props => {
    const { name, type, delivery_time, image } = props.hero;
    return (
        <View style={styles.container}>
            <Image style={styles.image} source={{ uri: image }} />
            <View style={styles.content}>
                <View style={styles.contentPrimary}>
                    <View style={styles.meta}>
                        <Text style={[styles.metaItem, styles.name]}>{name.toUpperCase()}</Text>
                        <Text style={[styles.metaItem, styles.type]}>
                            <MaterialIcons name="check" size={16} style={styles.check} /> Confirmed{' '}
                            {type}
                        </Text>
                    </View>
                    <TouchableOpacity style={styles.chatButton}>
                        <Image source={chatIcon} style={styles.chatImage} />
                    </TouchableOpacity>
                </View>
                <Text style={[styles.metaItem, styles.metaItemLast, styles.deliveryTime]}>
                    Estimated Delivery Time: {delivery_time} min
                </Text>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: '#fff',
        paddingBottom: emY(0.875),
        marginLeft: 27,
        marginRight: 27,
        borderBottomWidth: StyleSheet.hairlineWidth * 3,
        borderColor: Color.GREY_300
    },
    content: { flex: 1 },
    contentPrimary: { flexDirection: 'row' },
    meta: {
        flex: 1,
        marginTop: emY(1)
    },
    metaItem: {
        fontSize: emY(1),
        marginBottom: emY(0.4375)
    },
    metaItemLast: {
        marginBottom: 0
    },
    name: {},
    type: {},
    deliveryTime: {},
    check: {},
    image: {
        width: IMAGE_SIZE,
        height: IMAGE_SIZE,
        borderRadius: IMAGE_SIZE / 2,
        marginTop: emY(1),
        marginRight: 17
    },
    chatButton: {
        backgroundColor: Color.GREY_400,
        width: CHAT_SIZE,
        height: CHAT_SIZE,
        borderRadius: CHAT_SIZE / 2,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: emY(0.625)
    },
    chatImage: {
        width: CHAT_IMAGE_SIZE,
        height: CHAT_IMAGE_SIZE
    }
});

export default HeroDetail;
