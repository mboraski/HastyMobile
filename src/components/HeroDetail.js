// Third Part Imports
import React from 'react';
import { View, Image, StyleSheet, TouchableOpacity } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

// Relative Imports
import Text from './Text';
import Color from '../constants/Color';
import { emY } from '../utils/em';
import callIcon from '../assets/icons/call.png';
import logo from '../assets/profile.png';
// import logo from '../assets/icons/HastyOrangeIcon.png';

const PROFILE_IMAGE_SIZE = emY(3.75);
const CHAT_SIZE = emY(2.75);
const CHAT_IMAGE_SIZE = emY(1.25);

const HeroDetail = props => {
    const {
        firstName,
        lastName,
        deliveryTime,
        heroStatus,
        contractorId,
        contactContractor
    } = props;

    const callContractor = () => contactContractor(contractorId);

    const profileImage = props.profileImage || logo;

    return (
        <View style={styles.container}>
            <Image style={styles.profileImage} source={profileImage} />
            <View style={styles.content}>
                <View style={styles.contentPrimary}>
                    <View style={styles.meta}>
                        <Text style={[styles.metaItem, styles.name]}>
                            {firstName.toUpperCase()} {lastName.toUpperCase()}
                        </Text>
                        <Text style={[styles.metaItem, styles.link]}>
                            <MaterialIcons
                                name="check"
                                size={16}
                                style={styles.check}
                            />
                            {'Confirmed for order'}
                        </Text>
                        <Text style={[styles.metaItem, styles.deliveryTime]}>
                            Delivery Time: ~{deliveryTime / 60} min
                        </Text>
                        <Text
                            style={[
                                styles.metaItem,
                                styles.metaItemLast,
                                styles.deliveryTime
                            ]}
                        >
                            Status: {heroStatus}
                        </Text>
                    </View>
                </View>
            </View>
            <TouchableOpacity
                style={styles.chatButton}
                onPress={callContractor}
            >
                <Image source={callIcon} style={styles.chatImage} />
            </TouchableOpacity>
        </View>
    );
};

// <TouchableOpacity
//     onPress={() => {
//     }}
// >
//     <Text style={[styles.metaItem, styles.link]}>
//         <MaterialIcons name="check" size={16} style={styles.check} />{'Confirmed Order'}
//     </Text>
// </TouchableOpacity>

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: '#fff',
        paddingBottom: emY(0.875),
        marginLeft: 27,
        marginRight: 27,
        borderBottomWidth: StyleSheet.hairlineWidth * 3,
        borderColor: Color.GREY_300
    },
    content: { flex: 1 },
    contentPrimary: { flexDirection: 'row' },
    meta: {
        flex: 1,
        marginTop: emY(1)
    },
    metaItem: {
        fontSize: emY(1),
        marginBottom: emY(0.4375)
    },
    metaItemLast: {
        marginBottom: 0
    },
    link: {
        color: Color.DEFAULT
    },
    name: {},
    type: {},
    deliveryTime: {},
    check: {},
    chatButton: {
        backgroundColor: Color.GREY_400,
        width: CHAT_SIZE,
        height: CHAT_SIZE,
        borderRadius: CHAT_SIZE / 2,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: emY(0.7)
    },
    chatImage: {
        width: CHAT_IMAGE_SIZE,
        height: CHAT_IMAGE_SIZE
    },
    profileImage: {
        marginTop: emY(0.7),
        marginRight: emY(0.7),
        width: PROFILE_IMAGE_SIZE,
        height: PROFILE_IMAGE_SIZE,
        borderRadius: PROFILE_IMAGE_SIZE / 2
    }
});

export default HeroDetail;
