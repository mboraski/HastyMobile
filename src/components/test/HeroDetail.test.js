import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import HeroDetail from '../HeroDetail';

const initialState = {};

const hero = {
    id: 1,
    name: 'Jessica Morgan',
    type: 'Instant',
    delivery_time: 4,
    image: 'https://facebook.github.io/react/img/logo_og.png'
};

describe('HeroDetail', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<HeroDetail hero={hero} />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
