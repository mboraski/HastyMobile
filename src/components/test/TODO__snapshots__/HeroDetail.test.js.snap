// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroDetail renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 1.5,
      "borderColor": "#cccccc",
      "flexDirection": "row",
      "marginLeft": 27,
      "marginRight": 27,
      "paddingBottom": 14,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://facebook.github.io/react/img/logo_og.png",
      }
    }
    style={
      Object {
        "borderRadius": 34,
        "height": 68,
        "marginRight": 17,
        "marginTop": 16,
        "width": 68,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 16,
                "marginBottom": 7,
              },
              Object {},
            ]
          }
        >
          JESSICA MORGAN
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 16,
                "marginBottom": 7,
              },
              Object {},
            ]
          }
        >
          <Icon
            allowFontScaling={false}
            name="check"
            size={16}
            style={Object {}}
          />
           Confirmed
           
          Instant
        </Text>
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#b2b2b2",
            "borderRadius": 27,
            "height": 54,
            "justifyContent": "center",
            "marginTop": 10,
            "width": 54,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 21,
              "width": 21,
            }
          }
        />
      </TouchableOpacity>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
            "marginBottom": 7,
          },
          Object {
            "marginBottom": 0,
          },
          Object {},
        ]
      }
    >
      Estimated Delivery Time: 
      4
       min
    </Text>
  </View>
</View>
`;
